presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -73,6 +76,57 @@ public static GeometryTypeName valueOf(String type)
         }
     }
 
+    // Copy of com.esri.core.geometry.Interop.translateFromAVNaN
+    private static double translateFromAVNaN(double n)
+    {
+        return n < -1.0E38D ? (0.0D / 0.0) : n;
+    }
+
+    @Nullable
+    public static Envelope deserializeEnvelope(Slice shape)
+    {
+        if (shape == null) {
+            return null;
+        }
+        BasicSliceInput input = shape.getInput();
+
+        input.readInt(); // spatial reference ID
+
+        Envelope overallEnvelope = null;
+        while (input.available() > 0) {
+            int length = input.readInt();
+            ByteBuffer buffer = input.readSlice(length).toByteBuffer().slice().order(LITTLE_ENDIAN);
```

Comment:
Why do we need the `slice()` call here?

Reply:
We don't need it here. I copy-pasted that part from deserialize(). I'll remove it.